test/common/devCompileOptions.cmake

```diff
@@ -157,7 +157,24 @@ ELSE()
         IF(alpaka_ACC_GPU_HIP_ENABLE)
             LIST(APPEND alpaka_DEV_COMPILE_OPTIONS "-Wno-unused-command-line-argument")
         ENDIF()
-    # ICC
+
+        SET(IS_ICPX OFF)
+        IF(CMAKE_CXX_COMPILER_ID STREQUAL "IntelLLVM")
+            SET(IS_ICPX ON)
+        ELSEIF (CMAKE_VERSION VERSION_LESS 3.20 AND CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
+            INCLUDE(CheckCXXSourceCompiles)
+            CHECK_CXX_SOURCE_COMPILES("\
+                #ifndef __INTEL_LLVM_COMPILER\
+                #    error\
+                #endif
+                int main() {}"
+                IS_ICPX)
+        ENDIF()
+        IF (IS_ICPX)
+            # fast math is turned on by default with ICPX, which breaks our unit tests
+            LIST(APPEND alpaka_DEV_COMPILE_OPTIONS "-ffp-model=precise")
+        ENDIF()
```

Comment:
I eventually settled with distinguishing between clang and icpx and only enable fp model precise for icpx.